### PR TITLE
Add sdformat

### PR DESCRIPTION
**Pull request by joxoby**
Specify library name and version:  **sdformat/10.0.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by prince-chrismc** [comment]
I see a lot of the comments in #3215 will apply here as well.

Thank you for your **awesome** contributions :1st_place_medal:

File: recipes/sdformat/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+class SDFormat(ConanFile):
+    name = "sdformat"
+    license = "Apache-2.0"
+    homepage = "http://sdformat.org/"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "Simulation Description Format (SDFormat) parser and description files."
+    topics = ("ignition", "robotics", "simulation")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"fPIC": [True, False]}
```
Comment (madebr) [inline recipes/sdformat/all/conanfile.py:15]: Looks like sdformat only supports shared out-of-the box.
See https://github.com/osrf/sdformat/blob/144f82cf8b994abd2a566a7bbe6ae873a9d92403/cmake/SDFUtils.cmake#L56
Have you looked at patching it to support static builds?
Or is there interest upstream to add this possibility?

File: recipes/sdformat/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+    def requirements(self):
+        self.requires("ignition-math/6.4.0")
+        self.requires("tinyxml2/8.0.0")
+
+    def _configure_cmake(self):
+        self.cmake = CMake(self)
+        self.cmake.definitions["BUILD_TESTING"] = False
+        self.cmake.definitions["USE_INTERNAL_URDF"] = True
+        self.cmake.configure(source_folder=self._source_subfolder)
```
Comment (madebr) [inline recipes/sdformat/all/conanfile.py:73]:
```suggestion
        if self._cmake is not None:
            return self._cmake
        self._cmake = CMake(self)
        self._cmake.definitions["BUILD_TESTING"] = False
        self._cmake.definitions["USE_INTERNAL_URDF"] = True
        self._cmake.configure(source_folder=self._source_subfolder)
        return self._cmake
```

File: recipes/sdformat/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    homepage = "http://sdformat.org/"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "Simulation Description Format (SDFormat) parser and description files."
+    topics = ("ignition", "robotics", "simulation")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"fPIC": [True, False]}
+    default_options = {"fPIC": True}
+    generators = "cmake", "cmake_find_package_multi"
+
+    @property
```
Comment (madebr) [inline recipes/sdformat/all/conanfile.py:19]:
```suggestion
    generators = "cmake", "cmake_find_package_multi"

    _cmake = None

    @property
```

File: recipes/sdformat/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+    def _configure_cmake(self):
+        self.cmake = CMake(self)
+        self.cmake.definitions["BUILD_TESTING"] = False
+        self.cmake.definitions["USE_INTERNAL_URDF"] = True
+        self.cmake.configure(source_folder=self._source_subfolder)
+
+    def build(self):
+        self._configure_cmake()
+        self.cmake.build()
```
Comment (madebr) [inline recipes/sdformat/all/conanfile.py:77]:
```suggestion
        cmake = self._configure_cmake()
        cmake.build()
```

File: recipes/sdformat/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        self.cmake.install()
```
Comment (madebr) [inline recipes/sdformat/all/conanfile.py:81]:
```suggestion
        cmake = self._configure_cmake()
        cmake.install()
```

File: recipes/sdformat/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+        for dll_pattern_to_remove in ["concrt*.dll", "msvcp*.dll", "vcruntime*.dll"]:
+            for dll_to_remove in glob.glob(os.path.join(self.package_folder, "bin", dll_pattern_to_remove)):
+                os.remove(dll_to_remove)
+
+    def package_info(self):
+        version_major = self.version.split('.')[0]
+        version_till_minor = ".".join(self.version.split(".")[0:2])
+        self.cpp_info.libs = tools.collect_libs(self)
+        self.cpp_info.names["cmake_find_package"] = "sdformat{}".format(version_major)
```
Comment (madebr) [inline recipes/sdformat/all/conanfile.py:94]: In `package`, you remove `lib/pkgconfig`.
Can you also explicitly set the `pkg_config` generator name?
There is a real possibility that in the future, conan only generates `.pc` files for dependencies which explicitly sets the `pkg_config` property.

File: recipes/sdformat/all/test_package/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,13 @@
+cmake_minimum_required(VERSION 3.10.2)
```
Comment (madebr) [inline recipes/sdformat/all/test_package/CMakeLists.txt:1]: `3.10.2` because of `CXX_STANDARD 17`?

File: recipes/sdformat/all/conanfile.py (hunk 1 above)
Comment (joxoby) [reply recipes/sdformat/all/conanfile.py:15]: https://github.com/osrf/sdformat/pull/394

File: recipes/sdformat/all/test_package/CMakeLists.txt (hunk 7 above)
Comment (joxoby) [reply recipes/sdformat/all/test_package/CMakeLists.txt:1]: I'm just using the same as `sdformat` https://github.com/osrf/sdformat/blob/144f82cf8b994abd2a566a7bbe6ae873a9d92403/CMakeLists.txt#L1

**Pull request closed without merging**